src/inc-js-variants.php

```diff
@@ -42,120 +42,52 @@
-                        alt: 'Valid HGVS syntax!',
-                        title: 'Valid HGVS syntax!'
-                    }).show();
-                    // Check if the variant description is a c.? or a g.?. If it is, then do not let the user map the variant.
-                    if (oVariantDNA.val().substring(1,3) == '.?') {
-                        oVariantDNA.siblings('button:eq(0)').hide();
-                        oProtein.siblings('button:eq(0)').hide();
-                    } else if (!$.isEmptyObject(aTranscripts)) {
-                        // Only enable the mapping buttons when there are transcripts added to this variant.
-                        oVariantDNA.siblings('button:eq(0)').show();
-                        oProtein.siblings('button:eq(0)').show();
-                        // Hide possible 'view prediction button'
-                        $('#' + jq_escape(oProtein.attr('name')) + '_view_prediction').remove();
-                    }
-                }
-            });
-
-    } else if (bHGVS) {
-        // We didn't need Mutalyzer, and we know we've got a good-looking variant here.
-        oVariantDNA.siblings('img:first').attr({
-            src: 'gfx/check.png',
-            alt: 'Valid HGVS syntax!',
-            title: 'Valid HGVS syntax!'
-        }).show();
-        if (!$.isEmptyObject(aTranscripts)) {
-            // Only enable the mapping buttons when there are transcripts added to this variant.
-            oVariantDNA.siblings('button:eq(0)').show();
-            oProtein.siblings('button:eq(0)').show();
-            // Hide possible 'view prediction button'
-            $('#' + jq_escape(oProtein.attr('name')) + '_view_prediction').remove();
-        }
 
-    } else {
-        // No HGVS syntax, but no "real" onChange trigger yet, either.
-        oVariantDNA.siblings('img:first').hide();
-        if (!$.isEmptyObject(aTranscripts)) {
-            oVariantDNA.siblings('button:eq(0)').hide();
-            oProtein.siblings('button:eq(0)').hide();
-        }
-    }
-    return false;
+function lovd_checkHGVS(e)
+{
+    // This function should be called from any DNA variant input field
+    //  to check the HGVS of the given value. It sends the input (and
+    //  all other relevant data on the form) to
+    //  ajax/check_hgvs_dialogue.php. This script will then perform
+    //  the actual checks and return mapping information if possible.
+
+    var sVariant = $(this).val();
+    var sFieldName = $(this).attr("name");
+    var oChromosome = $('select[name="chromosome"]');
+    var sChromosome = (
+        oChromosome.length ? // Yes=VOG form; No=VOT form.
+            oChromosome.val() :
+            $("td:contains('Chromosome')").last().filter(function () {
+                return $(this).html() === "Chromosome";
+            }).siblings().eq(1).html()
+    );
+    var sRefSeqInfo = (
+        sFieldName[0] !== 'V'? // Yes=RefSeq is a transcript; No=RefSeq is genomic.
```

Comment:
I don't understand this logic? Don't all field names start with a 'V'?

Reply:
Transcriptfields are formatted as follows: <ID>_VariantOnTranscript/DNA. Genomic fields are formatted like this: 'VariantOnGenome/DNA/<ID>'. So the 'V' only returns true for genomic fields. I do agree that this could be done in a much cleaner way.

Reply:
Ahh, right! I forgot about the transcript ID prefix.
We could drop the comment if we do a search for "Transcript" in the field name, then the code makes it obvious.